via/services/secure_link.py

```diff
@@ -0,0 +1,39 @@
+from h_vialib.exceptions import TokenException
+from h_vialib.secure import ViaSecureURL
+from pyramid.httpexceptions import HTTPUnauthorized
+
+
+def has_secure_url_token(view):
+    """Require the request to have a valid signature."""
+
+    def view_wrapper(context, request):
+        if not request.find_service(SecureLinkService).request_is_valid(request):
+            return HTTPUnauthorized()
+
+        return view(context, request)
+
+    return view_wrapper
+
+
+class SecureLinkService:
+    def __init__(self, secret, signed_urls_required):
+        self._secure_token = ViaSecureURL(secret)
```

Comment:
```suggestion
        self._via_secure_url = ViaSecureURL(secret)
```

The name change from "Via Secure URL" to "secure token" confused me when reading the `request_is_valid()` method below. I had to look back up here at the `__init__()` to see what `self._secure_token` was and to see that it's an instance of `ViaSecureURL`.

The previous version of the code was more readable by doing `ViaSecureURL(request.registry.settings["via_secret"])).verify(request.url)` every time. No looking around to see what instance variables are or where the secret comes from. But I can see that this version avoids creating a new `ViaSecureURL` every time

Reply:
I've renamed the variable